sdks/go/sdk.go

```diff
@@ -139,14 +140,18 @@ func (s *SDK) WatchGameServer(f GameServerCallback) error {
 			gs, err = stream.Recv()
 			if err != nil {
 				if err == io.EOF {
-					_, _ = fmt.Fprintln(os.Stderr, "gameserver event stream EOF received")
+					log.Println("gameserver event stream EOF received")
 					return
 				}
-				_, _ = fmt.Fprintf(os.Stderr, "error watching GameServer: %s\n", err.Error())
+				log.Printf("error watching GameServer: %s\n", err.Error())
 				// This is to wait for the reconnection, and not peg the CPU at 100%.
 				time.Sleep(time.Second)
 				continue
 			}
+
+			if gs.ObjectMeta.DeletionTimestamp != 0 {
```

Comment:
🤔 

So I would handle this a bit differently. There are two problems we are trying to resolve:

1. Not output errors here when shutting down
2. Have a pluggable log system.

So, I would actually make the second part, part of the first - because we still want to handle `io.EOF` as above, but not output the log.

So I would have a pluggable log as per:
https://github.com/googleforgames/agones/issues/3245#issuecomment-1623999369

But have a function that is called here where the log functions are, that checks if `gs.ObjectMeta.DeletionTimestamp.isZero()` (I think that's the right function)` and if not, then runs the pluggable log function.

Does that make sense?

Reply:
I've included the Logger function to log the error message when `gs.ObjectMeta.DeletionTimestamp` is not zero and removed log when  "err == io.EOF" and any other error occurs while watching the GameServer.

`DeletionTimestamp` is of type int64, so I added the following condition:
 `if reflect.ValueOf(gs.ObjectMeta.DeletionTimestamp).IsZero()`  to check if it is zero or not. 
 [reference](https://stackoverflow.com/questions/33115946/how-to-know-if-a-variable-of-arbitrary-type-is-zero-in-golang)